configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/care/CareConfiguration.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.fhir.examples.configurablecare.care
+
+import android.content.Context
+import ca.uhn.fhir.context.FhirContext
+import ca.uhn.fhir.context.FhirVersionEnum
+import com.google.gson.Gson
+import com.google.gson.JsonArray
+import org.hl7.fhir.r4.model.Resource
+
+data class RequestResourceConfig(
+  var resourceType: String,
+  var values: List<Value>,
+  var maxDuration: String,
+  var unit: String
+) {
+  data class Value(var field: String, var value: String)
+}
+
+class ImplementationGuideConfig(
+  var implementationGuideId: String,
+  var entryPoint: String,
+  var requestResourceConfigurations: List<RequestResourceConfig>,
+  var supportedValueSets: JsonArray
+)
+
+data class SupportedImplementationGuide(
+  var location: String,
+  var carePlanPolicy: String,
+  var implementationGuideConfig: ImplementationGuideConfig,
+)
+
+data class CareConfiguration(var supportedImplementationGuides: List<SupportedImplementationGuide>)
+
+object ConfigurationManager {
+  private var careConfiguration: CareConfiguration? = null
+  private var taskConfigMap: MutableMap<String, String> = mutableMapOf()
```

Comment:
Why don't we directly use `RequestResourceConfig` from CareConfiguration? Why create another map in CareConfiguration ?

Reply:
These taskConfigMap and serviceRequestConfigMap are configs for current PlanDefinition (selected from spinner) right ?

I think this is not the right place for it. 
Check my other comment. What we can instead do is fetch current PlanDefinition's requestResourceConfig and pass that to re-instantiate the taskManager.

Reply:
There were some workarounds that I had done because CarePlanManager and TaskManager were originally in workflow. I think this is one of them. I'll fix it!